### ISSUE TITLE
Weapon "Press E to pickup" dialogue broken when "Switch Weapon" unbound

### ISSUE THREAD

**Issue by RossGosling-Github** [TTT-2/TTT2#697]
## Your version of TTT2

v0.7.4b

## Describe the bug

When I look at equipment such as a vanilla Deagle, M16, or Incendiary, the entire HUD disappears and the "Press E to pickup" dialogue does not appear, it also doesn't work when I press E to pickup.

## To reproduce

Steps to reproduce the behaviour:

1. Look at a weapon or grenade, I've been looking at the vanilla ones, M16, Deagle, Incendiary, etc
2. See error:
```
[TTT2 (Base) - v0.7.4b - New Roles, Extensible, Better Performance, ULX Support] gamemodes/terrortown/gamemode/client/cl_targetid.lua:713: bad argument #1 to 'upper' (string expected, got no value)
  1. upper - [C]:-1
   2. HUDDrawTargetIDWeapons - gamemodes/terrortown/gamemode/client/cl_targetid.lua:713
    3. Call - gamemodes/terrortown/gamemode/client/cl_targetid.lua:323
     4. unknown - gamemodes/terrortown/gamemode/client/cl_hud_manager.lua:93 (x291)
````

## Expected behaviour

You should be able to see the "Press E to pickup" dialogue over equipment and press E to pick them up

## Context (please provide as much as you can)

- The only workshop plugin is TTT2 v0.7.4b
- Screenshots: [https://imgur.com/a/LvPxGAO](https://imgur.com/a/LvPxGAO)

**Comment by TimGoll** [comment]
This is a known Bug that happens if you unbind your pickup key

**Comment by RossGosling-Github** [comment]
Just to snoop into the code, there seemingly a 'simple' but not necessarily the best solution would be here:
[/gamemodes/terrortown/gamemode/client/cl_targetid.lua#L718-L796](https://github.com/TTT-2/TTT2/blob/a1fb79d16f253208d4c91205e3746a4e0cd0b9b8/gamemodes/terrortown/gamemode/client/cl_targetid.lua#L718-L796)

Just bypassing the entire system if weapon pickup isn't bound because it'll never be relevant, alternatively there would be room to have a special unbound message or smth

**Comment by Histalek** [comment]
we definitely need to keep in mind what happens if `ttt_weapon_autopickup` is set to `0`. 
IMO we either need a big warning on the targetid if you have the pickupkey unbinded or even disable unbinding the pickupkey altogether.

**Comment by TimGoll** [comment]
@RossGosling-Github yes should be fixed easily. I tend to forget it.

@Histalek I'm confused. What has this convar todo with the weapon pickup?

**Comment by Histalek** [comment]
well if it is set to 0 and you have unbound your pickup key you can't pickup weapons at all, or am i missing something?

**Comment by TimGoll** [comment]
Yes. Should there really be a warning?

**Comment by Histalek** [comment]
that's open for discussion. i'm not saying there should be, but we should keep in mind that there is a way for the client to break their own gameplay

**Comment by ZenBre4ker** [comment]
Well, what I would like there to be would be a general "unbound"-key, that you get shown if it isnt binded.
And a warning clientsode could be nice. Maybe a little notification in the hud or in chat, that says, that you can't pickup weapons right now, please bind your key.

**Comment by TimGoll** [comment]
alright. I'll do something this weekend!

**Comment by TimGoll** [comment]
I will fix this as soon as #669 is merged to minimize merge conflict problems.
